### PR TITLE
Java: Don't clear content in store steps in summaries.

### PR DESCRIPTION
**Pull request by aschackmull**
Store steps in summaries aren't likely to be intended to clear the targeted content (also, this currently doesn't even work consistently). This caused false negative flow for any field-of-argument-to-callback summary as these summaries simply didn't work.

This fix could alternatively have been achieved by changing the synthesised read-/store-steps such that more store steps would be inferred as side-effects through reverse reads, but that seemed unnecessarily complicated.

### CODE REVIEW

**Review by hvitved — DISMISSED** [review]
LGTM

File: java/ql/lib/semmle/code/java/dataflow/internal/DataFlowPrivate.qll [hunk 1]
```diff
@@ -150,8 +150,7 @@ predicate readStep(Node node1, Content f, Node node2) {
  * in `x.f = newValue`.
  */
 predicate clearsContent(Node n, Content c) {
-  c instanceof FieldContent and
-  n = any(PostUpdateNode pun | storeStep(_, c, pun)).getPreUpdateNode()
+  exists(FieldAccess fa | instanceFieldAssign(_, fa) and n = getFieldQualifier(fa))
```
Comment (atorralba) [inline java/ql/lib/semmle/code/java/dataflow/internal/DataFlowPrivate.qll:153]: Isn't a problem that `c` is now unbound here?

Comment (hvitved) [reply java/ql/lib/semmle/code/java/dataflow/internal/DataFlowPrivate.qll:153]: Good spot!

Comment (aschackmull) [reply java/ql/lib/semmle/code/java/dataflow/internal/DataFlowPrivate.qll:153]: Good catch. That's a bug.

**Review by hvitved — APPROVED** [review]

**Pull request merged**